source/_integrations/tuya.markdown

```diff
@@ -15,123 +16,93 @@ ha_codeowners:
+**2.** Click `Link Tuya App Account` > `Add App Account`.
 
-    - Incorrect username or password: Enter the correct account and password of the Tuya Smart or Smart Life app in the **Account** and **Password** fields. Note that the app account depends on which app (Tuya Smart or Smart Life) you used to link devices on the [Tuya IoT Platform](https://iot.tuya.com/cloud/).
+![](/images/integrations/tuya/image_003.png)
 
-    - Incorrect availability zone: See [Availability Zone](https://github.com/tuya/tuya-home-assistant/wiki/Tuya-IoT-Platform-Configuration-Guide-Using-Smart-Home-PaaS#region--available-zone-correspondence) and select the correct availability zone.
+**3.** Scan the QR code that appears using the `Tuya Smart` app or `Smart Life` app.
 
-"1100:param is empty":
-  description: Empty parameter of username or app. Fill the parameters refer to [Tuya credential table](https://github.com/tuya/tuya-home-assistant/wiki/Install-Tuya-v2?_source=d10de34623e3daca5b02e3c31528a0c4#3-enter-your-tuya-credential).
+![](/images/integrations/tuya/image_004.png)
 
-"2406:skill id invalid":
-  description: Make sure that your cloud project on the [Tuya IoT Platform](https://iot.tuya.com/cloud/) should be created after May 25, 2021. Otherwise, you need to create a new project or migrate data to a new project. For more information, see [Operation on the Tuya IoT Platform](https://developer.tuya.com/en/docs/iot/migrate-from-an-older-version?id=Kamee9wtbd00b#title-3-Operation%20on%20the%20Tuya%20IoT%20Platform).
+**4.** Click `Confirm` on your `Tuya Smart` or `Smart Life` app.
 
-"28841105:No permissions. This project is not authorized to call this API":
-  description: >
-    Some APIs are not authorized, please [Subscribe](https://developer.tuya.com/en/docs/iot/applying-for-api-group-permissions?id=Ka6vf012u6q76#title-2-Subscribe%20to%20cloud%20products) then [Authorize](https://developer.tuya.com/en/docs/iot/applying-for-api-group-permissions?id=Ka6vf012u6q76#title-3-Authorize%20projects%20to%20call%20the%20cloud%20product). The following APIs must be subscribed for this tutorial:
+**5.** Navigate to the `All Devices` tab. Here you should be able to find the devices that have been added through your `Tuya Smart` or `Smart Life` account, as a confirmation everything worked.
 
-    - Authorization
+![](/images/integrations/tuya/image_005.png)
 
-    - Smart Home Devices Management
+#### Get authorization key
 
-    - Smart Home Family Management
+Click the created project to enter the `Project Overview` page and get the `Authorization Key` used to make API calls. You will need these for setting up the integration. in the next step.
 
-    - Smart Home Scene Linkage
+![](/images/integrations/tuya/image_006.png)
 
-    - Smart Home Data Service
+{% include integrations/config_flow.md %}
 
-    - Device status notification
+## Error codes and troubleshooting
 
-{% endconfiguration_basic %}
+|Error Code|Error Message|Troubleshooting|
```

Comment:
Tables suck, please revert this to the configuration_basic tag

Reply:
Apologies